controller/engine_controller.go

```diff
@@ -1676,3 +1684,30 @@ func (ec *EngineController) isResponsibleFor(e *longhorn.Engine, defaultEngineIm
 
 	return isPreferredOwner || continueToBeOwner || requiresNewOwner, nil
 }
+
+func EngineOpStatusFilter(e *longhorn.Engine) {
+	tcpReplicaAddrMap := map[string]bool{}
```

Comment:
Suggest to use `map[string]struct{}` instead.

Reply:
Can you explain more about this? Will Golang apply any optimizations when using `struct{}`?

Reply:
I was looking where the boolean is used, and seems we are only using it for key lookups. AFAIK Using an empty struct doesn’t require memory allocated but a bool does.